File: client/app/src/main/java/com/yszln/qiuqiu/ui/main/view/behavior/HeaderLogoBehavior.java
package com.yszln.qiuqiu.ui.main.view.behavior;

import android.content.Context;
import android.content.res.Resources;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import androidx.coordinatorlayout.widget.CoordinatorLayout;

import com.yszln.qiuqiu.R;


/**
 * @author LiBinBin
 * @create 2020/7/29
 * @Describe
 */
public class HeaderLogoBehavior extends CoordinatorLayout.Behavior<View> {


    public HeaderLogoBehavior(Context context, AttributeSet attrs) {
        super(context, attrs);

    }

    @Override
    public boolean layoutDependsOn(CoordinatorLayout parent, View child, View dependency) {
        if (dependency != null && dependency.getId() == R.id.image) {
            return true;
        }
        return false;
    }

    @Override
    public boolean onDependentViewChanged(CoordinatorLayout parent, View child, View dependency) {
        if (dependency != null && dependency.getId() == R.id.image) {
            Resources resources = dependency.getResources();

            final float progress = 1.f -
                    Math.abs(dependency.getTranslationY() / (dependency.getHeight() - resources.getDimension(R.dimen.collapsed_header_height)));
            TextView tv = child.findViewById(R.id.toolbar_tv);
            tv.setAlpha(progress);
            return true;
        }
        return false;

    }

}

File: client/app/src/main/java/com/yszln/qiuqiu/ui/main/view/behavior/HeaderLogoBehavior.kt
package com.yszln.qiuqiu.ui.main.view.behavior

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.TextView
import androidx.coordinatorlayout.widget.CoordinatorLayout
import com.yszln.qiuqiu.R
import kotlin.math.abs

class HeaderLogoBehavior(context: Context, attrs: AttributeSet) :
    CoordinatorLayout.Behavior<View>(context, attrs) {

    override fun layoutDependsOn(
        parent: CoordinatorLayout,
        child: View,
        dependency: View
    ): Boolean {
        return if (dependency != null && dependency.id == R.id.image) {
            true
        } else super.layoutDependsOn(parent, child, dependency)
    }

    override fun onDependentViewChanged(
        parent: CoordinatorLayout,
        child: View,
        dependency: View
    ): Boolean {
        return if (dependency != null && dependency.id == R.id.image) {
            val resources = dependency.resources
            val progress = 1f -
                    abs(dependency.translationY / (dependency.height - resources.getDimension(R.dimen.collapsed_header_height)))
            val tv = child.findViewById<TextView>(R.id.toolbar_tv)
            tv.alpha = progress
            true
        } else false
    }
}